File: buzu/src/test/java/br/net/buzu/sample/order/Customer.java
package br.net.buzu.sample.order;

import java.util.ArrayList;
import java.util.List;

public class Customer {
	
	private String name;
	private List<Address> addresses = new ArrayList<>();
	private List<String> phones;
	
	public Customer(String name, List<Address> addresses, List<String> phones) {
		super();
		this.name = name;
		this.addresses = addresses;
		this.phones = phones;
	}

	public Customer() {
	}
	
	public String getName() {
		return name;
	}
	public void setName(String name) {
		this.name = name;
	}
	public List<String> getPhones() {
		return phones;
	}
	public void setPhones(List<String> phones) {
		this.phones = phones;
	}

	public List<Address> getAddresses() {
		return addresses;
	}

	public void setAddresses(List<Address> addresses) {
		this.addresses = addresses;
	}
	
	public void addAddress(Address address){
		this.addresses.add(address);
	}

	@Override
	public int hashCode() {
		final int prime = 31;
		int result = 1;
		result = prime * result + ((addresses == null) ? 0 : addresses.hashCode());
		result = prime * result + ((name == null) ? 0 : name.hashCode());
		result = prime * result + ((phones == null) ? 0 : phones.hashCode());
		return result;
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Customer other = (Customer) obj;
		if (addresses == null) {
			if (other.addresses != null)
				return false;
		} else if (!addresses.equals(other.addresses))
			return false;
		if (name == null) {
			if (other.name != null)
				return false;
		} else if (!name.equals(other.name))
			return false;
		if (phones == null) {
			if (other.phones != null)
				return false;
		} else if (!phones.equals(other.phones))
			return false;
		return true;
	}

	@Override
	public String toString() {
		StringBuilder sb = new StringBuilder();
		sb.append(" name=").append(name).append("\n");
		sb.append(" addresses=").append(addresses).append("\n");
		sb.append(" phones=").append(phones);
		return sb.toString();
	}

}

File: ppl-lib/src/test/kotlin/br/net/buzu/sample/order/Customer.kt
package br.net.buzu.sample.order

import java.util.ArrayList

class Customer {

    var name: String? = null
    private var addresses: MutableList<Address>? = ArrayList()
    var phones: List<String>? = null

    constructor(name: String, addresses: MutableList<Address>, phones: List<String>) : super() {
        this.name = name
        this.addresses = addresses
        this.phones = phones
    }

    constructor() {}

    fun getAddresses(): List<Address>? {
        return addresses
    }

    fun setAddresses(addresses: MutableList<Address>) {
        this.addresses = addresses
    }

    fun addAddress(address: Address) {
        this.addresses!!.add(address)
    }

    override fun hashCode(): Int {
        val prime = 31
        var result = 1
        result = prime * result + if (addresses == null) 0 else addresses!!.hashCode()
        result = prime * result + if (name == null) 0 else name!!.hashCode()
        result = prime * result + if (phones == null) 0 else phones!!.hashCode()
        return result
    }

    override fun equals(obj: Any?): Boolean {
        if (this === obj)
            return true
        if (obj == null)
            return false
        if (javaClass != obj.javaClass)
            return false
        val other = obj as Customer?
        if (addresses == null) {
            if (other!!.addresses != null)
                return false
        } else if (addresses != other!!.addresses)
            return false
        if (name == null) {
            if (other.name != null)
                return false
        } else if (name != other.name)
            return false
        return if (phones == null) {
            other.phones == null
        } else
            phones == other.phones
    }

    override fun toString(): String {
        val sb = StringBuilder()
        sb.append(" name=").append(name).append("\n")
        sb.append(" addresses=").append(addresses).append("\n")
        sb.append(" phones=").append(phones)
        return sb.toString()
    }

}
